### PR TITLE
drivers/kernel_test: rename the kernel test driver

### PR DESCRIPTION
**Pull request by deepaksrma**
- rename the kernel_test_drv as os_test_drv because kernel_test contains kernel and network TCs and also may contain filesystem TCs.
- remove unnecessary header files

Signed-off-by: Deepak sharma <deepak.sh@samsung.com>

### CODE REVIEW

File: os/drivers/kernel_test/Make.defs [hunk 1]
```diff
@@ -16,10 +16,10 @@
 #
 ###########################################################################
 
-ifeq ($(CONFIG_DRIVERS_KERNEL_TEST),y)
+ifeq ($(CONFIG_OS_KERNEL_TEST),y)
 include kernel_test$(DELIM)kernel$(DELIM)Make.defs
 include kernel_test$(DELIM)network$(DELIM)Make.defs
-CSRCS += kernel_test_drv.c
+CSRCS += os_test_drv.c
```
Comment (kishore-sn) [inline os/drivers/kernel_test/Make.defs]: Maybe better to call it os_api_test_drv.c

File: os/include/tinyara/kernel_test_drv.h [hunk 2]
```diff
@@ -30,14 +30,14 @@
...
-#ifdef CONFIG_DRIVERS_KERNEL_TEST
+#ifdef CONFIG_OS_KERNEL_TEST
 
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
 /* Configuration
  *
- * CONFIG_DRIVERS_KERNEL_TEST - Enables kernel_test driver support
+ * CONFIG_OS_KERNEL_TEST - Enables kernel_test driver support
```
Comment (kishore-sn) [inline os/include/tinyara/kernel_test_drv.h]: Pls change comment description also

File: os/include/tinyara/kernel_test_drv.h [hunk 3]
```diff
@@ -30,14 +30,14 @@
 #include <tinyara/config.h>
 #include <tinyara/fs/ioctl.h>
 
-#ifdef CONFIG_DRIVERS_KERNEL_TEST
+#ifdef CONFIG_OS_KERNEL_TEST
```
Comment (kishore-sn) [inline os/include/tinyara/kernel_test_drv.h]: We will also need to rename this header file as os_api_test_drv.h

File: build/configs/artik053/STDK/defconfig [hunk 4]
```diff
@@ -436,7 +436,7 @@ CONFIG_ADC=y
 CONFIG_ADC_FIFOSIZE=8
 CONFIG_ADC_NPOLLWAITERS=2
 # CONFIG_DAC is not set
-# CONFIG_DRIVERS_KERNEL_TEST is not set
+# CONFIG_OS_KERNEL_TEST is not set
```
Comment (kishore-sn) [inline build/configs/artik053/STDK/defconfig]: CONFIG_OS_KERNEL_TEST still refers to kernel test. 
Better to change it as CONFIG_OS_API_TEST

File: os/drivers/kernel_test/os_test_drv.c [hunk 5]
```diff
@@ -27,8 +27,7 @@
 #include <tinyara/fs/fs.h>
 #include <tinyara/kernel_test_drv.h>
 #include <tinyara/sched.h>
-#include "signal/signal.h"
-#include "kernel_test_proto.h"
+#include "os_test_proto.h"
```
Comment (kishore-sn) [inline os/drivers/kernel_test/os_test_drv.c]: Also, need to rename all api's and data structures.

File: os/drivers/kernel_test/Make.defs (hunk 1 above)
Comment (deepaksrma) [reply os/drivers/kernel_test/Make.defs]: done

File: os/include/tinyara/kernel_test_drv.h (hunk 2 above)
Comment (deepaksrma) [reply os/include/tinyara/kernel_test_drv.h]: done

File: os/include/tinyara/kernel_test_drv.h (hunk 3 above)
Comment (deepaksrma) [reply os/include/tinyara/kernel_test_drv.h]: done

File: os/drivers/kernel_test/os_test_drv.c (hunk 5 above)
Comment (deepaksrma) [reply os/drivers/kernel_test/os_test_drv.c]: done

File: build/configs/artik053/STDK/defconfig (hunk 4 above)
Comment (deepaksrma) [reply build/configs/artik053/STDK/defconfig]: done

**Pull request closed without merging**